test/integration/controller_instance_test.go

```diff
@@ -0,0 +1,1475 @@
+					},
+				}
+				ct.osbClient.PollLastOperationReaction = &fakeosb.PollLastOperationReaction{
+					Response: &osb.LastOperationResponse{
+						State:       osb.StateFailed,
+						Description: strPtr("testDescription"),
+					},
+				}
+			},
+			skipVerifyingInstanceSuccess: true,
+			verifyCondition: v1beta1.ServiceInstanceCondition{
+				Type:   v1beta1.ServiceInstanceConditionReady,
+				Status: v1beta1.ConditionFalse,
+				Reason: "ProvisionCallFailed",
+			},
+		},
+		{
+			name: "last operation response successful with originating identity",
+			setupFunc: func(ct *controllerTest) {
+				if err := utilfeature.DefaultFeatureGate.Set(fmt.Sprintf("%v=true", scfeatures.OriginatingIdentity)); err != nil {
+					t.Fatalf("Failed to enable originating identity feature: %v", err)
+				}
+
+				ct.osbClient.ProvisionReaction = &fakeosb.ProvisionReaction{
+					Response: &osb.ProvisionResponse{
+						Async: true,
+					},
+				}
+				ct.osbClient.PollLastOperationReaction = &fakeosb.PollLastOperationReaction{
+					Response: &osb.LastOperationResponse{
+						State:       osb.StateSucceeded,
+						Description: strPtr("testDescription"),
+					},
+				}
+			},
+			verifyCondition: v1beta1.ServiceInstanceCondition{
+				Type:   v1beta1.ServiceInstanceConditionReady,
+				Status: v1beta1.ConditionTrue,
+				Reason: "ProvisionedSuccessfully",
+			},
+			preCreateInstance: func(ct *controllerTest) {
+				catalogClient, err := changeUsernameForCatalogClient(ct.catalogClient, ct.catalogClientConfig, "instance-creator")
+				if err != nil {
+					t.Fatalf("could not change the username for the catalog client: %v", err)
+				}
+				ct.catalogClient = catalogClient
+				ct.client = catalogClient.ServicecatalogV1beta1()
+
+			},
+			postCreateInstance: func(ct *controllerTest) {
+				verifyUsernameInLastBrokerAction(ct.t, ct.osbClient, fakeosb.PollLastOperation, "instance-creator")
+			},
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ct := &controllerTest{
+				t:                            t,
+				broker:                       getTestBroker(),
+				instance:                     getTestInstance(),
+				skipVerifyingInstanceSuccess: tc.skipVerifyingInstanceSuccess,
+				skipInstanceDelete:           tc.skipInstanceDelete,
+				setup:                        tc.setupFunc,
+				preDeleteBroker:              tc.preDeleteBroker,
+				preCreateInstance:            tc.preCreateInstance,
+				postCreateInstance:           tc.postCreateInstance,
+			}
+			ct.run(func(ct *controllerTest) {
+				if err := util.WaitForInstanceCondition(ct.client, testNamespace, testInstanceName, tc.verifyCondition); err != nil {
+					t.Fatalf("error waiting for instance condition: %v", err)
+				}
+			})
+		})
+	}
+}
+
+func TestAttemptProbablyShouldDelete(t *testing.T) {
+	ct := &controllerTest{
+		t:      t,
+		broker: getTestBroker(),
+		instance: func() *v1beta1.ServiceInstance {
+			i := getTestInstance()
+			i.Generation = 2
```

Comment:
You cannot set the Generation, the DeletionTimestamp, or the Status of a ServiceInstance when you create it. The API Server is going to clear out both of those fields. So, this test is just testing a simple provision and deprovision. I'm not very clear what you are trying to test here anyway.

Reply:
Yes, the test name was a reminder to myself to delete it. I was testing the API behavior as you pointed out, so I should have already removed it.